js/index.js

```diff
@@ -1,100 +1,360 @@
+        },{},{}
+      ],
+      rules:{//ruls for validation
+        title:webix.rules.isNotEmpty,
+        year:function(value){
+          return (value >= 1970 && value <= 2021);
         },
-        {
-           template:"The software is provided by <a href='https://webix.com'>https://webix.com</a>. All rights reserved &#169;",
-           css:"TextAlign",
-           height:30,
-           width:550
+        rating:function(value){
+          //console.log(value.replace(",", ".")); //for correct validation of float type
+          return (value.replace(",", ".") > 0 && value.replace(",", ".") <= 10);
+        },
+        votes:function(value){
+          if(value == 0 || (value.replace(",", ".") > 0 && value.replace(",", ".") < 100000)){
+            return value;
+          }else{return false}
         }
+      }
+      /*onValidationError:function(key, data){
+        webix.message.position = "bottom";
+        webix.message({text:key.toUpperCase()+"  field is incorrect", type:"error"});
+      }*/
+    };
+  var list = {
+    type:"clean",
+    rows:[
+      {
+        height: 35,
+        view:"toolbar",
+        elements:[
+          {
+            view:"text",
+            id:"list_input",
+            on:{
+              onTimedKeyPress:function(){
+                var value = this.getValue().toLowerCase();
+                $$("list").filter(function(obj){
+                    return obj.name.toLowerCase().indexOf(value)==0;
+                })
+              }
+            },
+          },
+          {view:"button", id:"btn_sort_asc", value:"Sort asc", css:"webix_primary", width:120,
+            click:function(){
+              webix.message({text:"Sort asc", expire:350});
+              $$("list").sort("#name#","asc","string");
+            }
+          },
+          {view:"button", id:"btn_sort_desc", value:"Sort desc", css:"webix_primary", width:120,
+            click:function(){
+              webix.message({text:"Sort desc", expire:350});
+              $$("list").sort("#name#","desc","string");
+            }
+          }
+        ]
+      },
+      {
+        view:"list",
+        css:"user_list",
+        id:"list",
+        height:300,
+        scrollY:true,
+        scrollX:false,
+        select:true,
+        template:"#name# <span class='remove_list_item_btn webix_icon mdi mdi-close'></span>",
+        onClick:{
+          remove_list_item_btn:function(e, id){
+            this.remove(id);
+            return false;
+          }
+        },
+        url:users_data_link
+      }
+    ]
+  };
+  var chart = {
+    type:"clean",
+    rows:[
+      {
+        view:"chart",
+        type:"bar",
+        value:"#age#",
+        label:"#age#",
+        minHeight:300,
+        tooltip:"User: #name#",
+        border:true,
+        barWidth:40,
+        radius:1,
+        xAxis:{
+            template:"#age#"
+        },
+        gradient:"falling",
+        url:users_data_link
+      },
+      {
+        view:"label",
```

Comment:
instead of a separate UI label, use `title` property of xAxis object

Reply:
Was modified!